saleor/core/notification/mutation_handler.py

```diff
@@ -0,0 +1,86 @@
+import json
+
+from ...account.models import User
+from ...core.permissions import AccountPermissions
+from ...core.permissions import OrderPermissions, ProductPermissions
+from ...order.models import Order
+from ...product.models import ProductVariant
+from ...webhook.payloads import (
+    generate_customer_payload,
+    generate_order_payload,
+    generate_product_variant_payload,
+)
+
+PAYLOAD_MAPPING = {
+    "ProductVariant": (
+        ProductVariant,
+        generate_product_variant_payload,
+        list,
+        ProductPermissions.MANAGE_PRODUCTS,
+    ),
+    "User": (
+        User,
+        generate_customer_payload,
+        None,
+        AccountPermissions.MANAGE_USERS,
+    ),
+    "Order": (
+        Order,
+        generate_order_payload,
+        None,
+        OrderPermissions.MANAGE_ORDERS,
+    ),
+}
+
+
+def get_payload_params(model_type):
+    if payload_params := PAYLOAD_MAPPING.get(model_type):
+        return payload_params
+
+
+def send_notification(manager, external_event_type, payloads, plugin_id=None):
+    if isinstance(payloads, list):
+        for payload in payloads:
+            trigger_notifications(manager, external_event_type, payload, plugin_id)
+    else:
+        trigger_notifications(manager, external_event_type, payloads, plugin_id)
+
+
+def trigger_notifications(manager, external_event_type, payload, plugin_id=None):
+    if plugin_id:
+        manager.notify_in_single_plugin(plugin_id, external_event_type, payload)
+    else:
+        manager.notify(event=external_event_type, payload=payload)
+
+
+class ExternalNotificationTriggerPayload:
```

Comment:
You don't need this class. It doesn't do anything except for calling a few methods and passing class attributes as params. We can easily refactor it as simple functions. First a few notes:

1. `pks` and `model` - these args are passed only to create a queryset inside the class. We can reduce the number of arguments, if we extracted the queryset inside the mutation, and passed it as a single argument. See the examples below.
2. `payload_inputs` - this name suggests that this is some kind of data dictionary, while this are objects from Django ORM - I renamed it to `objects` (could be also `queryset`).

```python
def _get_extracted_payload_input(object, payload_function, extra_payload):
    payload = json.loads(payload_function(object))
    payload = payload[0] if type(payload) == list else payload
    payload.update({"extra_payload": extra_payload})
    return payload


def get_external_notification_payload(
    objects, payload_function, input_type, extra_payload
):
    if input_type == list:
        payloads = json.loads(payload_function(objects))
        for payload in payloads:
            payload.update({"extra_payload": extra_payload})
        return payloads

    return [
        _get_extracted_payload_input(object, payload_function, extra_payload)
        for object in objects
    ]
```

In the mutation, we can use it as:
```python
objects = model.objects.filter(pk__in=pks)
payload = get_external_notification_payload(
    objects, payload_function, input_type, extra_payload
)
```

Something like that should work. What do you think? IMO this is much simpler. You really don't need classes most of the time.